### PR TITLE
Add functionality to detect spam bot join waves

### PR DESCRIPTION
**Pull request by The-personified-devil**
The exception handling/response for running a command with the has_role check whilst missing role is not yet implemented.

### CODE REVIEW

**Comment by The-personified-devil** [comment]
Borrowed the code (with some modifications) for the exception handling from https://gist.github.com/EvieePy/7822af90858ef65012ea500bcecf1612.

**Comment by The-personified-devil** [comment]
Made it back into a draft, because Lenny showed me a *lot* of issues that still need to be fixed.

File: src/cogs/error_handler.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @commands.Cog.listener()
+    async def on_command_error(self, ctx, error):
+        if hasattr(ctx.command, 'on_error'):
+            return
+
+        cog = ctx.cog
+        if cog:
+            if cog._get_overridden_method(cog.cog_command_error) is not None:
+                return
```
Comment (Yuhanun) [inline src/cogs/error_handler.py]: if cog and cog._get_...

File: src/cogs/rules.py [hunk 2]
```diff
@@ -37,6 +47,57 @@ async def snipe(self, ctx):
         content = message.content
         return await ctx.send(f"**{discord.utils.escape_markdown(discord.utils.escape_mentions(user))}** said on {ts} UTC:\n{content}")
 
+    async def _get_dm_channel(self, member):
+        dm_channel = member.dm_channel
+        if dm_channel == None:
+            dm_channel = await member.create_dm()
+        return dm_channel
```
Comment (Yuhanun) [inline src/cogs/rules.py]:
```suggestion
        return member.dm_channel or await member.create_dm()
```

You don't have to create a dm channel tho, discord.py internally already does this 🤷

File: src/cogs/rules.py [hunk 3]
```diff
@@ -37,6 +47,57 @@ async def snipe(self, ctx):
...
+
+    async def _notify_staff(self, guild, max_amount, message):
+        staff_role = guild.get_role(self.file["staff_role"]);
+        selected_staff = [
```
Comment (Yuhanun) [inline src/cogs/rules.py]: or just ping the staff role?...

**Review by Yuhanun — CHANGES_REQUESTED** [review]

**Review by Mehodin — APPROVED** [review]

**Pull request merged**